File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    const filteredStr = s.toLowerCase().replace(/[^a-z0-9]/g, '');

    const reversedStr = filteredStr.split('').reverse().join('');

    return filteredStr === reversedStr
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        if (s == " ") return true;
        s = s.toLowerCase();
        String result = s.replaceAll("[^a-z0-9]", "");
        String reverse = new StringBuilder(result).reverse().toString(); 
 
        System.out.printf("값 출력: result = %s, reverse = %s%n", result, reverse);
        return result.equals(reverse);
    }
}